conf/site.conf

```diff
@@ -71,6 +71,17 @@ IMAGE_BOOT_FILES_append_sama5d27-som1-ek-sd = "\
 "
 WKS_FILE_sama5d27-som1-ek-sd = "yoe-sdimage.wks"
 
+PREFERRED_PROVIDER_qemu-native ?= "qemu-native"
+PREFERRED_PROVIDER_nativesdk-qemu ?= "nativesdk-qemu"
+
+# Xilinx zcu102 specific configurations
+IMAGE_BOOT_FILES_append_zcu102-zynqmp = " boot.bin"
+PREFERRED_PROVIDER_qemu-native_zcu102-zynqmp = "qemu-xilinx-native"
+PREFERRED_PROVIDER_nativesdk-qemu_zcu102-zynqmp = "nativesdk-qemu-xilinx"
+PREFERRED_PROVIDER_virtual/pmu-firmware_zcu102-zynqmp = "pmu-firmware"
+PREFERRED_PROVIDER_virtual/boot-bin_zcu102-zynqmp = "xilinx-bootbin"
+PREFERRED_PROVIDER_virtual/dtb_zcu102-zynqmp = "device-tree"
```

Comment:
that looks better, as options are only set for Xilinx machines. thoughts @kraj ?

Reply:
if these options are only being used for single machine `zcu102-zynqmp` then better place for this is machine config file but if its SOC family settings then it makes sense here.

Reply:
@kraj do you mean in meta-xilinx/meta-xilinx-bsp/conf/machine/zcu102-zynqmp.conf ? But that would be a pull request for the xilinx giuthub.

Reply:
right

Reply:
> if these options are only being used for single machine `zcu102-zynqmp` then better place for this is machine config file but if its SOC family settings then it makes sense here.

@kraj 
I haven't tried other SoC's from Xilinx but they may require this change too, I don't want to assume without trying. On a separate note, is wildcard possible here ? Something like.
PREFERRED_PROVIDER_virtual/dtb_*-zynqmp = "device-tree"

Reply:
I don't think wildcard work in such cases and its also not a good idea, since you want to be particular about it.

Reply:
> right

I looked a bit closer, it looks like Xilinx sets PREFERRED_PROVIDER_virtual/pmu-firmware, PREFERRED_PROVIDER_virtual/boot-bin and PREFERRED_PROVIDER_virtual/dtb in their petalinux distro
https://github.com/Xilinx/meta-petalinux/blob/master/conf/distro/include/petalinux-features.conf#L34

So I tried to build using DISTRO=petalinux bitbake petalinux-image-minimal
but the petalinux distro is including the poky distro.
https://github.com/Xilinx/meta-petalinux/blob/master/conf/distro/petalinux.conf#L1

however poky distro isn't being provided by Yoe workspace in my understanding.
Not sure how to proceed, kindly advise 

If build with petalinux distro then then following variable don't need to be set.
PREFERRED_PROVIDER_virtual/pmu-firmware, PREFERRED_PROVIDER_virtual/boot-bin and PREFERRED_PROVIDER_virtual/dtb

Reply:
I'm not sure best how to handle the poky requirement ... it is my opinion that distros should not require poky -- that severely limits the portability of the layer. 

This brings up a more basic question -- should we support multiple distros in the Yoe directory structure? I don't see any reason offhand why we can't -- simply remove meta-yoe from the layers -- however, we then loose many of the nice yoe defaults ...

Reply:
these are all machine specific settings, so they should either be in machine config file or perhaps user specific `local.conf `